apps/dashboard/app/models/allowlist_policy.rb

```diff
@@ -17,7 +17,10 @@ def initialize(allowlist)
   # @raises ArgumentError if any allowlist path or permitted? argument
   #         has the form ~user/some/path where user doesn't exist
   def permitted?(path)
-    allowlist.blank? || allowlist.any? { |parent| child?(Pathname.new(parent), real_expanded_path(path.to_s)) }
+    key = path_to_key(path)
+    Rails.cache.fetch(key) do
+      allowlist.blank? || allowlist.any? { |parent| child?(Pathname.new(parent), real_expanded_path(path.to_s)) }
```

Comment:
This looks like it is checking a condition that would be true always in that isn't it asking "if the allowlist is blank or has anything in it?" which seems to cover the 2 states it could be in, so it would always be true. Is that right? I'm thinking i'm missing something here.

Reply:
>  isn't it asking "if the allowlist is blank or has anything in it?" 

It's more like `if the allowlist is blank or has anything in it that matches this criteria?`

Which allows for the 3rd state, that is it `has something, but nothing that matches that criteria`.

Reply:
Would it make more sense to use `permitted?` then? Isn't that the same logic?

Reply:
I didn't know you could pass a block like that to `any?`. Makes sense now.